### ISSUE TITLE
Dense Layer 2 not considered Output Layer 

### ISSUE THREAD

**Issue by rammadhav987** [deeplearning4j/deeplearning4j#3673]
This is the error I am getting Can anyone guide on the Output Layer?
https://gist.github.com/rammadhav987/8209d3444d5c606d1936ab8f03fbb87f

the json file of my model
https://gist.github.com/rammadhav987/4c64d2530498ab70b130765960960802

**Comment by turambar** [comment]
@rammadhav987 So first, machine learning 101: you can't train a neural network (or any model, really) without am objective (loss, likelihood, whatever) function. In not so many words, that's what your error message means: you're trying to train without having defined your loss.

In DL4J, you add a loss function by adding a layer that implements the [IOutputLayer interface](https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/api/layers/IOutputLayer.java) -- the means the layer returns a "score" (loss). The only layers that satisfy those are LossLayers (example [here]()), which compute losses, or OutputLayers, which combine a standard layer with a loss function. Example of output layer [here](https://github.com/deeplearning4j/dl4j-examples/blob/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/feedforward/mnist/MLPMnistTwoLayerExample.java#L83).

Note that this differs from Keras, where loss functions are specified in the call to "compile" (see [here](https://github.com/fchollet/keras/blob/master/examples/imdb_cnn.py#L68)) and are NOT stored as a part of the model configuration JSON. The Keras loss function is stored as JSON in the "training_config" property of a full model file (stored as HDF5 format).

Thus, if you seek to import **and train** a Keras model, you must do one of two things:

- save your full Keras model (not just the config). See this [FAQ answer](https://keras.io/getting-started/faq/#how-can-i-save-a-keras-model) for instructions.
- import the Keras model config from JSON, then manually add the loss/output layer to the configuration yourself. That's somewhat involved -- you might consider leveraging our [transfer learning API](https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/transferlearning/TransferLearning.java).

**Comment by turambar** [comment]
@tomthetrainer @maxpumperla FYI

**Comment by turambar** [comment]
@rammadhav987 one last remark: read your error more closely. It has nothing to do with the output layer or your model -- there's a formatting problem with your data!
